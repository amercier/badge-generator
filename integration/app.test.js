const puppeteer = require('puppeteer');
const { port } = require('../jest-puppeteer.config').server;

let browser;
let page;

beforeEach(async () => {
  browser = await puppeteer.launch({
    dumpio: true,
    args: ['--no-sandbox', '--disable-setuid-sandbox'],
  });
  page = await browser.newPage();
  await page.emulate({
    viewport: {
      width: 1920,
      height: 1080,
    },
    userAgent: 'Puppeteer',
  });
  await page.goto(`http://localhost:${port}/`);
});

afterEach(async () => {
  await page.close();
  await browser.close();
});

test('Setting repository and enabling some services', async () => {
  await expect(page).toMatch('Github Badge Generator');

  // Set repository
  await page.focus('input[type="text"]');
  await page.keyboard.type('amercier/badge-generator', {
    delay: 20,
  });

  // Select services
  await page.click('label[title="Travis CI"]');
  await page.click('label[title="Codecov"]');
  await page.click('label[title="David DM"]');
  await page.click('label[title="David DM (dev)"]');

  const output = await page.evaluate(
    () => document.querySelector('pre').innerText,
  );
  expect(output).toMatchSnapshot();
});
