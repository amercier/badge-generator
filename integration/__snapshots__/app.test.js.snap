// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting repository and enabling some services 1`] = `
"[![Build Status](https://img.shields.io/travis/amercier/badge-generator/master.svg?style=default)](https://travis-ci.org/amercier/badge-generator)
[![Test Coverage](https://img.shields.io/codecov/c/github/amercier/badge-generator/master.svg?style=default)](https://codecov.io/github/amercier/badge-generator?branch=master)
[![Dependency Status](https://img.shields.io/david/amercier/badge-generator.svg?style=default)](https://david-dm.org/amercier/badge-generator)
[![devDependency Status](https://img.shields.io/david/dev/amercier/badge-generator.svg?style=default)](https://david-dm.org/amercier/badge-generator#info=devDependencies)
"
`;
